### PR TITLE
fix(KUI-1343): fix header visibility bug

### PR DESCRIPTION
**Pull request by allazis**
PR including fix for bugs described in [Jira issue KUI-1343](https://kth-se.atlassian.net/jira/software/c/projects/KUI/boards/45?assignee=712020%3A5ac3ffad-3dc6-4419-a9eb-25ac831231f6&selectedIssue=KUI-1343).

An attempt to  gather logics for whether to include a heading or not in Editor vs Preview have been made by placing help funcitons for this in the file `EditorAndPreviewUtils.js`. Unit tests for these functions are found [here](https://github.com/KTH/kurs-pm-data-admin-web/blob/fix/KUI-1343-fix-header-visibility-bug/public/js/app/util/__test__/EditorAndPreviewUtils.test.js). Test setup can be found in the following file attached in this PR and in Jira: 
[Test setup and results.xlsx](https://github.com/user-attachments/files/16635798/Test.setup.and.results.xlsx)

Subsections are now being listed as standard sections in `fieldsByType.js`. This, to resemble the way in which they are handled in the repo for viewing published course memos: `kurs-pm-web`.

If possible, please test bug fix manually by following the instructions below.

## **Test instructions**

**1st check**

- [ ] Open a new draft and stay at first tab with name _Innehåll och lärandemål_ 
- [ ] Scroll down to heading _Detaljplanering_ and fill in content **without checking the checkbox to include in course memo**. Save content.
- [ ] Go to preview and check that the heading _Detaljplanering_ or its content is **not** included in the course memo preview.

**2nd check**
- [ ] Go back to editing mode and go to second tab with name _Förberedelser inför kursstart_ 
- [ ] Scroll down to heading _Stöd för studenter med funktionsnedsättning_ and fill in content in its editable subsection **without checking the checkbox to include in course memo**. Save content.
- [ ] Go to preview and check that the content for the subsection for _Stöd för studenter med funktionsnedsättning_ is **not** included in the course memo preview

**3rd check**
- [ ] Go back to editing mode and include both heading _Detaljplanering_ and subsection for _Stöd för studenter med funktionsnedsättning_ by checking their checkboxes
- [ ] Go to preview and see that both are included

**4th check**
- [ ] Go back to editing mode and exclude both heading _Detaljplanering_ and subsection for _Stöd för studenter med funktionsnedsättning_ by unchecking their checkboxes
- [ ] Go to preview and see that both are **not** included

### Visibility for `"defaultTrue"`
Some headings are set to have visibility `"defaultTrue"` in the database. First time shown, these headings are checked:

1. If they have content, the visibility will be set to `true`. 
2. If they do not have content but are editable , their visibility will be set to `false`. This, to let the user choose whether to not include or not later
3. If they are not editable but empty, the value `"defaultTrue"` will be kept. 

### Visibility for `mandatoryForSome`
Some headings have the type set to `mandatoryForSome`. If there is content, these headings will be shown.

### CODE REVIEW

**Review by karlandindrakryggen — COMMENTED** [review]
Really good job! Just one small comment for now. 

I'll wait with approve until you looked at the bug that I messaged you about.

File: public/js/app/util/EditorAndPreviewUtils.js
[the remark this answers is not in the document]
Comment (allazis) [reply public/js/app/util/EditorAndPreviewUtils.js:1]: Agree!

**Comment by allazis** [comment]
> Really good job! Just one small comment for now.
> 
> I'll wait with approve until you looked at the bug that I messaged you about.

@karlandindrakryggen Thanks a lot! 

Bugfix pushed for when subsection was included from start: [fix(KUI-1343): add handling of editable sections that have visibility…](https://github.com/KTH/kurs-pm-data-admin-web/pull/379/commits/db9c27530bb0aac76feb82937a50be8f952a3d9c) Had a fix ready which I didn't sync on Friday apparently..

Please test again and confirm.

**Review by amirhossein-haerian — APPROVED** [review]
LGTM, reviewing this PR was insightful. thanks for the clear description and instructions. 
Well done ⭐️🚀

**Comment by allazis** [comment]
> LGTM

Thank you!

**Review by karlandindrakryggen — APPROVED** [review]
Bra jobbat 🎉

**Comment by karlandindrakryggen** [comment]
> > Really good job! Just one small comment for now.
> > I'll wait with approve until you looked at the bug that I messaged you about.
> 
> @karlandindrakryggen Thanks a lot!
> 
> Bugfix pushed for when subsection was included from start: [fix(KUI-1343): add handling of editable sections that have visibility…](https://github.com/KTH/kurs-pm-data-admin-web/pull/379/commits/db9c27530bb0aac76feb82937a50be8f952a3d9c) Had a fix ready which I didn't sync on Friday apparently..
> 
> Please test again and confirm.

I've tested it again now, and it works as it should now :)

**Pull request merged**